### DISCUSSION TITLE
Door lock functionality (fingerprints?)

### DISCUSSION THREAD

**Discussion by scrampker** [make-all/tuya-local#2110] [General]
I was just curious as to what features we should be seeing when integrating locks. I have some elcheapo fingerprint door handle lock, and I tried various devices that were listed, and really all I see is the status. There's no control/configuration other than lock/unlock, which doesn't work due to read-only error. Despite changing the lock to 'controllable' in the Tuya cloud item.

Are we supposed to be able to program codes/users/fingerprints, or anything else? IE, am I just using the wrong device type, or is that the full functionality to expect?

Thanks!

**Comment by make-all** [comment]
In general, security locks don't provide simple remote lock/unlock functionality. There is maybe a way to remotely unlock using a passcode, but the way to do that is not well understood so not implemented by this integration.  Adding new credentials is likewise not well understood, and not implemented.  Other config options that the lock offers can generally be made available, and the status - what method and credential id was used to open the lock should be available.

**Reply by mprobber** [reply]
How do I see what method and credential ID was used to unlock the lock in home assistant?

**Reply by make-all** [reply]
If available, it is in the `changed_by` attribute of the lock, which normally will show in the LogBook.

**Comment by scrampker** [comment]
So if configuration changes are impossible, is the only point so that you can base triggers off the sensor data?

**Comment by scrampker** [comment]
I haven't seen this data anywhere on my stuff.

-------- Original Message --------

**Comment by s0ftice** [comment]
I got my fingerprint Tuya BLE lock to work (as in trigger unlock via Home Assistant) 🎉
See details here https://github.com/make-all/tuya-local/issues/2059#issuecomment-2693126630
